1.15.2 support

Uraxys opened the pull request:
Prøvde meg på å oppdatere pluginen til 1.15.2, er ikke 100% ferdig ennå, men alt fungerer (At least the parts I tested), men vil ha feedback / om det er noe jeg har glemt.

Edit:
Changes that needs to be done to the database:
The 'logg' table needs to change itemID to varchar.
The 'blocklog' table needs to change material to varchar, the old blocklogs will still work, as I made a class that converts IDs to material, but it will say (LEGACY) next to it, as some materials might be changed in a update (double smooth stone slabs)

Uraxys committed:
1.15

Uraxys committed:
Fixed missing letter in pom.

Had a little oppsi while closing IntelliJ, I'm just happy git stores stashed code, but yeah, this is to fix that small oppsi.

Uraxys committed:
More protection & legacy support

Tested and it works, not done, but getting there.

Uraxys committed:
Forgot some material IDs in Sandtak

hultberg commented on src/main/java/no/minecraft/Minecraftno/listener/MinecraftnoPlayerListener.java:
```diff
@@ -321,18 +322,24 @@ private void adduser(Player player) {
+        // then it just cancels the event, even if it isn't an right click / left click event.
+        ItemStack itemInHand = player.getInventory().getItemInMainHand();
+        if (event.getHand() != null && event.getHand() == EquipmentSlot.HAND) itemInHand = player.getInventory().getItemInMainHand();
+        else if (event.getHand() == EquipmentSlot.OFF_HAND) player.getInventory().getItemInOffHand();
```
This change will make all special items for mods/admins work in both hands, do we want that? For instance allow adminsticking with your off hand, should that be allowed? Igniting stuff should be disallowed based on the config for both hands ofc.

Uraxys replied:
Might not be a good idea, at least from my experience, will change it.

Uraxys replied:
After taking a second look then I've already fixed that, if you look a bit down, then you see that I check if the hand equals main hand.
Line: 325, 441, 493, 509, 542, 561

mgrytbak commented:
Feel free to pop by our Discord-server and poke me there and i will add you to a hidden channel for the server, so we can arrange testing and so on if/when its ready. https://discord.gg/y8zmaen

Uraxys committed:
Removed a lot of deprecated methods.

Tested it locally and worked, but should be tested again, as I might have jumped over something.

Uraxys committed:
Forgot to commit a class

Uraxys committed:
Added a file with the needed sql changes.

Uraxys committed:
Update dbchange.sql

Uraxys committed:
Fixed the sponge (NOT TESTED)
